pkg/watch/mandatory_template_change.go

```diff
@@ -0,0 +1,68 @@
+package watch
+
+import (
+	"context"
+
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+)
+
+type MandatoryTemplateChangeHandler struct {
+	client.Reader
+	record.EventRecorder
+}
+
+func NewMandatoryTemplateChangeHandler(handlerClient ChangeHandlerClient) *MandatoryTemplateChangeHandler {
+	return &MandatoryTemplateChangeHandler{Reader: handlerClient, EventRecorder: handlerClient}
+}
+
+func (h *MandatoryTemplateChangeHandler) Watch() handler.MapFunc {
+	return func(ctx context.Context, o client.Object) []reconcile.Request {
+		requests := make([]reconcile.Request, 0)
+		template := &v1beta2.ModuleTemplate{}
+
+		if err := h.Get(ctx, client.ObjectKeyFromObject(o), template); err != nil {
+			return requests
+		}
+
+		if !template.Spec.Mandatory {
+			return nil
+		}
+
+		kymas := &v1beta2.KymaList{}
+		listOptions := &client.ListOptions{
+			LabelSelector: k8slabels.SelectorFromSet(k8slabels.Set{shared.ManagedBy: shared.OperatorName}),
+		}
+		err := h.List(ctx, kymas, listOptions)
+		if err != nil {
+			return requests
+		}
+
+		templateName := types.NamespacedName{
+			Namespace: template.GetNamespace(),
+			Name:      template.GetName(),
+		}
+		for _, kyma := range kymas.Items {
```

Comment:
same as this `Watch()` function, think about some code can be shared with `(h *TemplateChangeHandler) Watch()`.

for example, to append `requests`, basically you can define a function like `enqueueChangedKyma( kyma []v1beta2.kyma), then it can be shared.

And we already know basically log message not help us in most case, please don't introduce more log, instead, think about use events or put into resources status. For this case, I suggest remove this info log, together with the one from `(h *TemplateChangeHandler) Watch()`

Reply:
I removed the logger. But atm there is not any relevant code that can be shared. For example the append you mentioned is a one liner; do you think for `requests = append(requests, reconcile.Request{NamespacedName: kymaName})` we should have a function? But this is just a normal list append.

We can have a talk what you would like to have simplified